test/lit/wat-kitchen-sink.wast

```diff
@@ -1,43 +1,47 @@
 ;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
 
-;; RUN: wasm-opt --new-wat-parser -all %s -S -o - | filecheck %s
+;; RUN: wasm-opt --new-wat-parser --hybrid -all %s -S -o - | filecheck %s
 
 (module $parse
  ;; types
- ;; CHECK:      (type $s0 (struct ))
+ ;; CHECK:      (type $s0 (struct_subtype  data))
 
- ;; CHECK:      (type $void (func))
+ ;; CHECK:      (type $void (func_subtype func))
  (type $void (func))
- ;; CHECK:      (type $many (func (param i32 i64 f32 f64) (result anyref (ref func))))
+
+ ;; CHECK:      (type $many (func_subtype (param i32 i64 f32 f64) (result anyref (ref func)) func))
  (type $many (func (param $x i32) (param i64 f32) (param) (param $y f64)
                    (result anyref (ref func))))
 
- (type $s0 (struct))
+ (sub (type $s0 (struct)))
```

Comment:
Reading this, I'm not actually sure what "sub" does or means..?

Reply:
The reason there is no supertype here is that syntactically the supertypes are a vector, so it is allowed for there to be zero supertypes.